pkg/lvm/lvm.go

```diff
@@ -143,23 +144,30 @@ func (lvm *Lvm) Run() error {
 	return nil
 }
 
-func mountLV(lvname, mountPath string, vgName string) (string, error) {
+func mountLV(lvname, mountPath string, vgName string, fsType string) (string, error) {
 	lvPath := fmt.Sprintf("/dev/%s/%s", vgName, lvname)
 
 	formatted := false
+	if fsType == "" {
+		fsType = "ext4"
+	}
 	// check for already formatted
 	cmd := exec.Command("blkid", lvPath)
 	out, err := cmd.CombinedOutput()
 	if err != nil {
 		klog.Infof("unable to check if %s is already formatted:%v", lvPath, err)
 	}
-	if strings.Contains(string(out), "ext4") {
+	matches := regexp.MustCompile(`TYPE="(\w+)"`).FindStringSubmatch(string(out))
+	if len(matches) > 1 {
 		formatted = true
 	}
+	if len(matches) > 1 && matches[1] != fsType {
```

Comment:
Check for second condition can be inside upper if

Reply:
fixed